AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -696,58 +834,148 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
+        /// <param name="conId">The contract Id</param>
+        /// <returns>The single position PnL update event from TWS.</returns>
+        public Task<PnLSingleEventArgs> RequestPnLSingle(
+            string accountCode,
+            string modelCode,
+            int conId)
+        {
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestPnLSingle(accountCode, modelCode, conId, tokenSource.Token);
+        }
+
+        /// <summary>
+        /// Get the PnL of the account.
+        /// </summary>
+        /// <param name="accountCode">The account code</param>
+        /// <param name="modelCode">The model code</param>
+        /// <param name="conId">The contract Id</param>
+        /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
+        /// <returns>The single position PnL update event from TWS.</returns>
+        public Task<PnLSingleEventArgs> RequestPnLSingle(
+            string accountCode,
+            string modelCode,
+            int conId,
+            CancellationToken cancellationToken)
+        {
+            PnLSingleEventArgs pnlSingleEvent = null;
+            var taskSource = new TaskCompletionSource<PnLSingleEventArgs>();
+            EventHandler<PnLSingleEventArgs> pnlSingleEventHandler = null;
+
+            pnlSingleEventHandler = (sender, args) =>
             {
-                this.twsCallbackHandler.SecurityDefinitionOptionParameterEvent -= securityDefinitionEventHandler;
-                this.twsCallbackHandler.SecurityDefinitionOptionParameterEndEvent -= securityDefintionEndEventHandler;
-                taskSource.TrySetResult(securityDefinitionEvents);
+                pnlSingleEvent = args;
+                taskSource.TrySetResult(pnlSingleEvent);
             };
 
+            this.twsCallbackHandler.PnLSingleEvent += pnlSingleEventHandler;
+
             cancellationToken.Register(() =>
             {
                 taskSource.TrySetCanceled();
             });
 
-            this.twsCallbackHandler.SecurityDefinitionOptionParameterEvent += securityDefinitionEventHandler;
-            this.twsCallbackHandler.SecurityDefinitionOptionParameterEndEvent += securityDefintionEndEventHandler;
-
             int requestId = this.twsRequestIdGenerator.GetNextRequestId();
-            this.clientSocket.RequestSecurityDefinitionOptionParameters(requestId, underlyingSymbol, exchange, underlyingSecType, underlyingConId);
+            this.clientSocket.RequestPnLSingle(requestId, accountCode, modelCode, conId);
+
             return taskSource.Task;
         }
 
+        /// <summary>
+        /// Request PnL update cancelation
+        /// </summary>
+        /// <param name="reqId">The request Id</param>
+        public void CancelPnLSingle(int reqId)
+        {
+            this.clientSocket.CancelPnLSingle(reqId);
```

Comment:
use `this.twsRequestIdGenerator.GetNextRequestId()` to generate the `reqId` and remove it from the parameter

Reply:
Following the documentation https://interactivebrokers.github.io/tws-api/pnl.html#portfolio_window_pnl it seems we need to pass the same reqId as the one provided when requesting the PnL.